bpmn-model/src/main/java/io/zeebe/model/bpmn/validation/zeebe/EventBasedGatewayValidator.java

```diff
@@ -82,4 +96,16 @@ private boolean isValidEvent(final IntermediateCatchEvent event) {
           .anyMatch(e -> e.isAssignableFrom(eventDefinition.getClass()));
     }
   }
+
+  private Stream<MessageEventDefinition> getMessageEventDefinitions(
+      Collection<SequenceFlow> outgoingSequenceFlows) {
+    return outgoingSequenceFlows
+        .stream()
+        .map(SequenceFlow::getTarget)
+        .filter(t -> t instanceof IntermediateCatchEvent)
```

Comment:
Wouldn't an event based gateway only have outgoing flows to `IntermediateCatchEvent`?